src/components/views/messages/TextualBody.js

```diff
@@ -37,28 +38,76 @@ module.exports = React.createClass({
 
         /* link URL for the highlights */
         highlightLink: React.PropTypes.string,
+
+        /* callback for when our widget has loaded */
+        onWidgetLoad: React.PropTypes.func,
+    },
+
+    getInitialState: function() {
+        return {
+            link: null,
```

Comment:
It can't be set before the component is mounted, as the links in the message are calculated by linkifyElement() which has to run after the DOM has been created and the component mounted.  Therefore we'd still need to re-render() in order to pull in the right LinkPreviewWidget for the links we've discovered - and so might as well still be considered component state. wdyt?